From 7601e7e981fc127e920cba717fc1561d1fe9e353 Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Thu, 31 May 2018 11:10:55 +0300
Subject: [PATCH 3836/5242] MLK-16784-3: arm64: dts: add pdm nodes in iMX8MM
 dts

commit  8fdcb764cb818d15b10c29a0751904b13f4031f5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add micfil DAI node in dtsi and pdm sound card in dts.
We also  moved ak5558 nodes into separate dts since
ak5558 uses sai5 which share some pins with micfil.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
(cherry picked from commit 8451c6886b0175b7e1391293aa9fb461395f8485)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../boot/dts/freescale/fsl-imx8mm-evk-ak5558.dts   |   32 ++++++++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts   |   30 +++++++++++++++++-
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi      |    4 +--
 4 files changed, 65 insertions(+), 4 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak5558.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 9fee984..d1cac86 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -78,4 +78,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-ddr3l-arm2.dtb \
 				 fsl-imx8mq-evk-drm.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += fsl-imx8mm-evk.dtb \
 				 fsl-imx8mm-evk-ak4497.dtb \
-				 fsl-imx8mm-evk-m4.dtb
+				 fsl-imx8mm-evk-m4.dtb \
+				 fsl-imx8mm-evk-ak5558.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak5558.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak5558.dts
new file mode 100644
index 0000000..0268b09
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-ak5558.dts
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mm-evk.dts"
+
+/ {
+	sound-ak5558 {
+		status = "okay";
+	};
+	sound-micfil {
+		status = "disabled";
+	};
+};
+
+&micfil {
+	status = "disabled";
+};
+
+&sai5 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index d3a86e7..6872ead 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -89,6 +89,7 @@
 		model = "ak5558-audio";
 		audio-cpu = <&sai5>;
 		audio-codec = <&ak5558>;
+		status = "disabled";
 	};
 
 	sound-ak4497 {
@@ -106,6 +107,12 @@
 		spdif-out;
 		spdif-in;
 	};
+
+	sound-micfil {
+		compatible = "fsl,imx-audio-micfil";
+		model = "imx-audio-micfil";
+		cpu-dai = <&micfil>;
+	};
 };
 
 &clk {
@@ -215,6 +222,18 @@
 			>;
 		};
 
+		pinctrl_pdm: pdmgrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SAI5_MCLK_SAI5_MCLK	0xd6
+				MX8MM_IOMUXC_SAI5_RXC_PDM_CLK		0xd6
+				MX8MM_IOMUXC_SAI5_RXFS_SAI5_RX_SYNC	0xd6
+				MX8MM_IOMUXC_SAI5_RXD0_PDM_DATA0	0xd6
+				MX8MM_IOMUXC_SAI5_RXD1_PDM_DATA1	0xd6
+				MX8MM_IOMUXC_SAI5_RXD2_PDM_DATA2	0xd6
+				MX8MM_IOMUXC_SAI5_RXD3_PDM_DATA3	0xd6
+			>;
+		};
+
 		pinctrl_spdif1: spdif1grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SPDIF_TX_SPDIF1_OUT	0xd6
@@ -664,7 +683,7 @@
 	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
 	assigned-clock-rates = <0>, <49152000>;
 	fsl,sai-asynchronous;
-	status = "okay";
+	status = "disabled";
 };
 
 &spdif1 {
@@ -796,3 +815,12 @@
 &vpu_h1 {
 	status = "okay";
 };
+
+&micfil {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pdm>;
+	assigned-clocks = <&clk IMX8MM_CLK_PDM_SRC>, <&clk IMX8MM_CLK_PDM_DIV>;
+	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
+	assigned-clock-rates = <0>, <24576000>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index e00abeb..d2a1cec 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -661,7 +661,7 @@
 	};
 
 	micfil: micfil@30080000 {
-		compatible = "fsl,imx9mq-micfil";
+		compatible = "fsl,imx8mm-micfil";
 		reg = <0x0 0x30080000 0x0 0x10000>;
 		interrupts = <GIC_SPI 44 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 45 IRQ_TYPE_LEVEL_HIGH>,
@@ -670,7 +670,7 @@
 		clocks = <&clk IMX8MM_CLK_PDM_IPG>,
 			 <&clk IMX8MM_CLK_PDM_ROOT>;
 		clock-names = "ipg_clk", "ipg_clk_app";
-		dmas = <&sdma2 24 24 0>;
+		dmas = <&sdma2 24 26 0x80000000>;
 		dma-names = "rx";
 		status = "disabled";
 	};
-- 
1.7.9.5

